website_sharing_whole_class_mosaic_plot_translation.Rmd

```diff
@@ -0,0 +1,507 @@
+ 
+ 
+For graders and Professor: this document is intended as a tutorial in Chinese on the content of Mosaic plots with R. We combined texts from three websites, wiki, *Mosaic plots with ggplot2* (https://cran.r-project.org/web/packages/ggmosaic/vignettes/ggmosaic.html) and the **Chart: Mosaic** section on [edav.info](https://edav.info/). 
+
+We hope this document can effectively jumpstart any user (with limited language background to Chinese) with sufficient skills to assess Mosaic plots with R.
+
+## This document is outlined as follows:
+
+1. We first introduce the defination of the mosaic plot and its usage: what mosaic plots are, what features they have, what aesthetics they have. (This section is adapted from wiki website)
+2. We then describe in detail what the basic parts of plots are. (This section is adapted from edav.info)
+3. Finally, we walked through the conditions representation in mosaic plots, multiple variables in mosaic plots, and orders of variables from *Mosaic plots with ggplot2*.
+
+
+## 1. Introduction 介绍 (Mosaic plot 2019)
+
+马赛克图(也称马里梅科图)是一种将两个或两个以上定性变量进行可视化的统计报告图[1]，它是自旋图的多维扩展，自旋图只对一个变量的信息进行展示[2]。它给出数据的整体概况，并让识别变量之间的关系更加简单。例如，当不同类别的方框都具有相同的面积时，表示变量之间是独立的[3]。Hartigan和Kleiner于1981年创建了马赛克图，Friendly于1994年对其进行了扩展[4]。由于马赛克图很像马里梅科印刷品，因此也被称为梅科图。这是非常好的进行定性变量数据可视化的方式之一。
+
+与柱状图和自旋图一样，每个方框的面积，与该类别内的观测值数量成正比[5]。
+
+## 2.Basic Parts of Mosaic Plots
+
+本页是一个正在进行中的工作。我们感谢您的任何意见。如果你想帮助改善这个页面，请考虑向我们的repo贡献。
+
+### 2.1 综概
+
+```{r}
+library(readr)
+library(vcd)
+df = read_csv("/Users/hanxiaozhang/Downloads/MusicIcecream.csv")
+```
+
+马赛克图需要一些时间来学习如何进行正确的阅读和绘制。特别是在刚开始的时候，我们建议循序渐进的学习绘制：先从划分一个变量开始，然后每次添加一个其他的变量。完整的马赛克图会对每个变量都进行一次拆分。
+
+* 需要注意：如果你的数据有一个列是频率，就像下面的例子一样，计数列必须称为Freq。(表格Tables和矩阵matrices也可以使用，更多细节请参见?vcd::structable。)
+
+* 还要注意的是：所有这些图都是用vcd::mosaic()绘制的，而不是用R里的基本函数包mosaicplot()。
+
+
+### 2.2 数据如下。
+
+```{r data}
+df
+```
+
+### 2.3 划分变量
+
+#### 2.31 仅按年龄划分。
+
+```{r}
+vcd::mosaic(~Age, df)
+```
+
+#### 2.32 先按年龄划分，再按照音乐。
+
+```{r}
+vcd::mosaic(Music ~ Age, df)
+```
+
+需要注意的是，第一组在"年轻 "和 "年长”中拆分，第二组拆分则将每个年龄段的人按照"古典 "和 "摇滚"进行再次划分。
+
+#### 2.33 先按年龄划分，再按音乐划分，再按喜爱划分。
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, df)
+```
+
+### 2.34 划分的方向
+
+请注意，在前面的例子中，分割的方向如下。
+
+#### 2.341 年龄--横向分割
+
+#### 2.342 音乐--垂直分割
+
+#### 2.343 最喜欢的--横向分割
+
+这是默认的方向模式：交替方向从水平方向开始。因此，我们得到如下同样的图：
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, 
+       direction = c("h", "v", "h"), df)
+```
+
+可以根据需要改变方向。例如，要创建一个doubledecker图，除了最后一个变量，所有的分割都是垂直的。
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music,
+       direction = c("v", "v", "h"), df)
+```
+
+需要注意的是，方向向量是按照拆分的顺序（年龄、音乐、喜爱），而不是按照公式中变量出现的顺序。在公式中，最后一个要拆分的变量要在”~”前先列出来。
+
+
+### 2.4 填充颜色
+
+填充颜色的分类是根据最后划分的维度——即因变量（本例中最喜欢的冰淇淋口味）。(如果不能正常运行，请更新到vcd的最新版本。)
+
+```{r}
+vcd::mosaic(Favorite ~ Age + Music, 
+       highlighting_fill = c("grey90", "cornflowerblue"),
+       df)
+```
+
+### 2.5 标签
+
+想要看关于标签选项的官方文档，请参见Strucplot框架中关于标签的说明。
+
+#### 2.51 旋转标签
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.